rediscluster/nodemanager.py

```diff
@@ -65,12 +65,30 @@ def _validate_host_port_remap(self, host_port_remap):
             if not isinstance(item, dict):
                 raise RedisClusterConfigError("items inside host_port_remap list must be of dict type")
 
+            if len(set(item.keys()) - {'from_host', 'from_port', 'to_host', 'to_port'}) != 0:
+                raise RedisClusterConfigError("Invalid keys provided in host_port_remap rule")
+
             # If we have from_host, we must have a to_host option to allow for translation to work
             if ('from_host' in item and 'to_host' not in item) or ('from_host' not in item and 'to_host' in item):
-                raise RedisClusterConfigError("Both from_host and to_host must be present in remap item if either is defined")
+                raise RedisClusterConfigError("Both from_host and to_host must be present in host_port_remap rule if either is defined")
+
+            if ('from_port' in item and 'to_port' not in item) or ('from_port' not in item and 'to_port' in item):
+                raise RedisClusterConfigError("Both from_port and to_port must be present in host_port_remap rule if either is defined")
 
-            if ('from_port' in item and 'to_port' not in item) or  ('from_port' not in item and 'to_port' in item):
-                raise RedisClusterConfigError("Both from_port and to_port must be present in remap item")
+            try:
+                socket.inet_aton(item.get('from_host', '0.0.0.0').strip())
+                socket.inet_aton(item.get('to_host', '0.0.0.0').strip())
+            except socket.error:
+                raise RedisClusterConfigError("Both from_host and to_host in host_port_remap rule must be a valid ip address")
```

Comment:
Host doesn't have to be an IP address like in case of ElastiCache endpoints.

Reply:
Thats a good point, but the redis servers themselves return ips even in Elasticache(going by a test I did). So enforcing using ip addresses will be helpful in catching misconfigurations on that front. In the test that I did with Elasticache, I converted everything to ips and added that in the configuration
 

Having functionality to detect host names and work with host names may be a good feature but IMO may also have edge cases and issues during implementation and (again IMO) may be best done in a separate change and Pull Request

Reply:
Technically, i could acctually be in facor of implementing two different keywords in the source mapping to chose what kind of remapping you want to have, think like `from_host_ip` and `from_hostname` that would have two different paths through the code. It would not require some magic logic to determine what mode to use and the user can explicitly decide what mode to use and we can implement two different remapping algorithms in each case really. But i would say that this is something that is not super needed to be implemented into this PR right now but can be an addition later on.

Reply:
Yes Agreed.

Reply:
@mrnom I was thinking about that part about remapping hostnames instead of IP:s makes less sense then to have remapping IP:s. I was thinking that in the AWS ElasticCache case ain't it the purpose of the endpoint manager to handle the IP:s and hostnames to sort them for you. I would love for anyone to actually find a valid use-case where hostname remapping acctually makes any sense in reality over just to add it just to have it but there is no need for it.